security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+  VM Runner exectutes the derivation pipeline as part of the kona fault proof program running ontop of asterisc.
+  [Extensive test coverage](https://app.codecov.io/gh/anton-rs/kona/tree/main/crates) of kona.
+
+- **Detection:**
+  The VM runner would pick up an issue.
+  The action tests or local testing may pick up an issue.
+  Use of the derivation elsewhere, for example in alternative consensus clients like `op-rs`, would pick up an issue.
+
+- **Recovery Path(s):**
+  Fix the derivation pipeline divergence, release and deploy the updated kona + asterisc when in the coldpath.
+  If `kona` + `asterisc` is placed in the hotpath, either the `op-program` + `cannon` would need to be falled back on otherwise fallback to the permissed dispute game.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Kona Execution Diverges from op-program
+
+- **Description:**
+  The Kona Fault Proof Program executes blocks in a stateless L2 block executor that uses optimism-revm as a backend.
+  If optimism-revm or the L2 block executor diverge in behavior from the `op-program`, the `kona` + `asterisc` proof system will be invalid.
+
+- **Risk Assessment:**
+  Low likelihood, low impact.
+  A small divergence in execution behavior can cause invalid game results.
+  But since kona + asterisc is in the coldpath, it doesn't have impact on anything in the production system.
+
+- **Mitigations:**
+  State testing against op-revm.
+  Action tests covering the L2 block executor are run on every pull request in the [kona] repository.
+  `op-reth` which uses the same op-execution environment as `kona`.
+
+- **Detection:**
+  Running `op-reth` so it's synced and following tip would provide a way to watch for chain splits.
+  A chain split occuring means `kona`'s execution (the same as `op-reth`) diverges from the `op-program`.
+
+- **Recovery Path(s):**
+  If `kona` + `asterisc` is in the coldpath, it would need to be fixed and re-deployed.
+  If `op-program` + `cannon` cannot be placed in the hotpath, and `kona` + `asterisc` fails, the permissioned fallback must be engaged.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Breaking Backwards Compatibility for the op-challenger
+
+- **Description:**
+  The `op-challenger` has been updated to support the Kona + Asterisc Fault Dispute Game type ([Game Type 3]).
+
+- **Risk Assessment:**
+  Low likelihood as the action tests that are run in CI very frequently run the `op-challenger` with the `kona` + `asterisc` fault dispute game.
+  Medium impact since the off-chain `op-challenger` service can be updated with enough buffer time due to the large game time and nature of the chess clock.
```

Comment:
Could you also confirm, that if the `op-challenger` is broken through bug. Honest other actors will attack the malicious game to get the free money (bond). 
So even if our challenger is down, and there is sufficient honest actor in the wild this is not suppose to be an issue correct?

Reply:
Yes. This is the exact same model as the `op-program` + `cannon`, if the challenger goes down.